lib/verify/response.rb

```diff
@@ -84,14 +84,31 @@ def full_name
     end
 
     def most_recent_verified_value(attribute_name, required: true)
-      attrs = required ? attributes.fetch(attribute_name) : attributes.dig(attribute_name)
-      return if attrs.blank?
-
-      most_recent_verified_attribute = attrs.sort_by { |attribute| attribute["from"].present? ? Date.strptime(attribute["from"], "%Y-%m-%d") : 0 }
-        .reverse
-        .find { |attribute| attribute["verified"] }
-      raise MissingResponseAttribute, "No verified value found for #{attribute_name}" if required && most_recent_verified_attribute.nil?
-      required ? most_recent_verified_attribute.fetch("value") : most_recent_verified_attribute&.fetch("value")
+      if required
```

Comment:
Let's pair on trying to simplify the code in this method. There are now a fair few more explicit `return` and `raise` calls than before, plus more conditionals and I reckon we can get them down.